src/utilities/time.ts

```diff
@@ -96,3 +96,87 @@ export function getUnixEpochTime(doyTimestamp: string): number {
 
   return 0;
 }
+
+const usPeryear = 3.154e13;
+const usPerDay = 8.64e10;
+const usPerHour = 3.6e9;
+const usPerMinute = 6e7;
+const usPerSecond = 1000000;
+const usPerMillisecond = 1000;
+
+const hoursPerDay = 24;
+const minutesPerHour = 60;
+const secondsPerMinute = 60;
+
+export function convertUsToDurationString(durationUs: number): string {
+  const years = Math.floor(durationUs / usPeryear);
+  durationUs -= years * usPeryear;
+
+  const days = Math.floor(durationUs / usPerDay);
+  durationUs -= days * usPerDay;
+
+  const hours = Math.floor(durationUs / usPerHour) % hoursPerDay;
+  durationUs -= hours * usPerHour;
+
+  const minutes = Math.floor(durationUs / usPerMinute) % minutesPerHour;
+  durationUs -= minutes * usPerMinute;
+
+  const seconds = Math.floor(durationUs / usPerSecond) % secondsPerMinute;
+  durationUs -= seconds * usPerSecond;
+
+  const milliseconds = Math.floor(durationUs / usPerMillisecond);
+  durationUs -= milliseconds * usPerMillisecond;
+
+  const microseconds = durationUs;
+
+  return `${years}y ${days}d ${hours}h ${minutes}m ${seconds}s ${milliseconds}ms ${microseconds}us`;
+}
+
+export function convertDurationStringToUs(durationString: string): number | never {
```

Comment:
- Please add a descriptive docstring with example.
- Alphabetize function in file.